From 753d082a3c1c52d7470ea49133fa25865c0f85f0 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Thu, 1 May 2025 06:24:04 +0200
Subject: [PATCH 2/2] gcompris: fix absolute path

---
 pkgs/games/gcompris/default.nix | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/pkgs/games/gcompris/default.nix b/pkgs/games/gcompris/default.nix
index 60d87033272f..a393faabfdc8 100644
--- a/pkgs/games/gcompris/default.nix
+++ b/pkgs/games/gcompris/default.nix
@@ -27,6 +27,13 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-E3l+5l4rsauidl6Ik5gSWf+SGpVaAVi5X51etl1D05w=";
   };
 
+  # fix concatenation of absolute paths like
+  # /nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.0/bin/..//nix/store/77zcv3vmndif01d4wh1rh0d1dyvyqzpy-gcompris-25.0/share/gcompris-qt/rcc/core.rcc
+  postPatch = ''
+    substituteInPlace src/core/config.h.in  --replace-fail \
+      "../@_data_dest_dir@" "../share/gcompris-qt"
+  '';
+
   cmakeFlags = [
     (lib.cmakeFeature "QML_BOX2D_LIBRARY" "${qmlbox2d}/${qtbase.qtQmlPrefix}/Box2D.2.1")
     (lib.cmakeBool "BUILD_TESTING" finalAttrs.finalPackage.doCheck)
-- 
2.49.0

