From 3b88c1b1b9ec0abe79f291e82afaac0344ff7617 Mon Sep 17 00:00:00 2001
From: Guilhem Saurel <guilhem.saurel@laas.fr>
Date: Mon, 12 Jan 2026 20:13:52 +0100
Subject: [PATCH] Fix typing passwords starting with dashes

---
 src/rofi_rbw/typer/wtype.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/rofi_rbw/typer/wtype.py b/src/rofi_rbw/typer/wtype.py
index 41a47ddf87f3d7d39bba92210a8eb1e477203fc2..928f8ab7870ed3250c2aa5057eeeaf082d79de27 100644
--- a/src/rofi_rbw/typer/wtype.py
+++ b/src/rofi_rbw/typer/wtype.py
@@ -24,6 +24,7 @@ def type_characters(self, characters: str, start_delay: float, key_delay: int, a
         if key_delay > 0:
             args = args + ["-d", str(key_delay)]
 
+        args.append("--")
         args.append(characters)
         run(args)
 
